cpp/purify/yaml-parser.cc

```diff
@@ -28,10 +30,44 @@ YamlParser::YamlParser (const std::string& filepath)
 
 void YamlParser::readFile ()
 {
-  YAML::Node config = YAML::LoadFile(this->filepath_);
-  // A bit of defensive programming
-  assert(config.Type() == YAML::NodeType::Map);
-  this->config_file = config;
+  try
+    {
+      YAML::Node config = YAML::LoadFile(this->filepath_);
+      // A bit of defensive programming
+      assert(config.Type() == YAML::NodeType::Map);
+      this->config_file = config;
+    }
+  catch(YAML::BadFile& exception)
+    {
+      char cwd[1024];
+      getcwd(cwd, sizeof(cwd));
+      std::cerr << "The input file path " << this->filepath_ << " could not be found from " << cwd << std::endl;
```

Comment:
rather than using std::cerr put this in the runtime_error

Reply:
Also, why are you using a raw array of type `char`? Why not a std::string?

Reply:
That's what `getcwd` takes as argument.